[Low Siang Ern] iP

siangernlow opened the pull request.

Commit:
Add Gradle support

siangernlow committed:
Renamed Duke to TaskBot

User input functionality is now handled in the Main class, while TaskBot has a class by itself which implements functionality required for Level-1.

siangernlow committed:
Added a class to handle tasks

siangernlow committed:
TaskBot now supports adding tasks and listing them

The "echo" functionality of TaskBot has been replaced by the adding of tasks.

siangernlow committed:
Modified the user interface to use enums

Commands sent to TaskBot are now controlled by enums instead of strings.

siangernlow committed:
Improved TaskBot functionality and personality

TaskBot is now able to complete tasks according to user input. The personality of TaskBot is also more consistent throughout all the responses given.

siangernlow committed:
Made Task abstract, implemented three types of tasks

Deadlines, events and todos are concrete implementations of the now abstract Task class.

siangernlow committed:
TaskBot now handles todos, deadlines and events

siangernlow committed:
Modified logo of Taskbot

siangernlow committed:
Setup automated Text UI Testing

Updated the test paths of the .bat file (windows) to the correct paths, and updated input.txt to test all functionality of the bot.

siangernlow committed:
Added exceptions for invalid user input

siangernlow committed:
A new exception for completed tasks

siangernlow committed:
TaskBot now handles invalid or missing user inputs.

TaskBot can now detect if the format of the command is wrong, or user input is missing or invalid. TaskBot will then print a message describing the error.

siangernlow committed:
TaskBot can now delete tasks

The user can now delete tasks and appropriate error handling for deleting tasks has also been added

siangernlow committed:
Updated test files

siangernlow committed:
Insert source files into packages

Source files are located in the same directory and not categorised
into packages.

Moving the source files into relevant packages follows the
SE-EDU Java coding standard

siangernlow committed:
Revert "Insert source files into packages"

This reverts commit 6ca4e7872209f48924ef85bf15fbe363d0326ab0.

siangernlow committed:
Revert "Revert "Insert source files into packages""

This reverts commit 64b5100521fe51cac549292280ca022efcb892cf.

siangernlow committed:
Refactor classes to accomodate improvements

Each class is doing too much.

Refactor the code to make the program more modular and flexible to
changes.

siangernlow committed:
Add storage functionality to TaskBot

The user's list was not being stored and lost upon revisiting the
program.

Storing the list is vital for a todo program as we cannot expect
the user to keep the program switched on indefinitely.

siangernlow committed:
Require time format for deadline and events

There is no requirement for the format of the dates and times that
could be input for deadlines and events.

We should enforce a set format so that we can ensure the time
given is valid.

siangernlow committed:
Merge branch 'branch-Level-7'

siangernlow committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/taskbot/logic/Taskbot.java
#	src/taskbot/main/Main.java
#	src/taskbot/task/Deadline.java
#	src/taskbot/task/Event.java

siangernlow committed:
Make the code more OOP

The program has classes performing too many things themselves.

We should extract functionality from these classes into individual
classes, which also fulfils the iP considerable of MoreOOP.

siangernlow committed:
Make task inheritance more logical

Deadline and Event tasks both have time based parameters of
different names.

We should unify the classes to signify the
similarities between them.

siangernlow committed:
Refactor code to facilitate JUnit testing

siangernlow committed:
Implement JUnit tests

siangernlow committed:
Update miscellaneous file from JAR conversion

siangernlow committed:
Add find functionality to TaskBot

The user is unable to search for tasks based on keywords.

Being able to search for tasks using keywords will provide a
better user experience.

siangernlow committed:
Update JavaDocs of all files

siangernlow committed:
Enforce coding standards

siangernlow committed:
Merge branch 'branch-Level-9'

siangernlow committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/taskbot/task/TaskList.java

siangernlow committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/taskbot/command/DeleteCommand.java
#	src/taskbot/command/DoneCommand.java
#	src/taskbot/command/EventCommand.java
#	src/taskbot/command/TodoCommand.java
#	src/taskbot/command/UpcomingCommand.java
#	src/taskbot/main/Main.java
#	src/taskbot/task/TaskList.java

siangernlow committed:
Merge remote-tracking branch 'origin/add-gradle-support'

siangernlow committed:
Rename source files for Gradle

siangernlow committed:
Add Gradle support to TaskBot

siangernlow committed:
Merge branch 'branch-A-Gradle'

siangernlow committed:
Add Checkstyle support

siangernlow committed:
Implement Checkstyle regulations

The files do not adhere to certain Checkstyle rules and should be
modified to do so.

siangernlow committed:
Merge branch 'branch-A-Checkstyle'

siangernlow committed:
Use varargs for Find command

The find command can only support searching a single keyword.

We should extend the functionality to search for a variable number
of keywords.

siangernlow committed:
Merge branch 'branch-A-Varargs'

siangernlow committed:
Add JavaFX and FXML support

siangernlow committed:
Move UI output from console to application

Taskbot completes a command and outputs the result to the console.

As we have already implemented JavaFX, we should refactor the code
so that the output is displayed in the application instead.

siangernlow committed:
Integrate TaskBot into JavaFX application

siangernlow committed:
Merge branch 'branch-Level-10'

bobbylaw commented on src/main/java/taskbot/ui/DialogBox.java:
```diff
@@ -0,0 +1,67 @@
+        getChildren().setAll(tmp);
+        setAlignment(Pos.TOP_LEFT);
+    }
+
```
Should there be a header comment since the accessibility is public? (line 58)

siangernlow replied:
Hi, I believe that you are right in that there should be header comments for the public methods, and will resolve them shortly. Thanks for your time and feedback!

bobbylaw reviewed (commented):
Looks good, just the method header comment.
Keep up the good work

siangernlow committed:
Update JavaDoc comments

underthehai commented on src/main/java/taskbot/command/Command.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    /**
+     * Performs the relevant operation tied to the Command type on the given task list.
```
I think this part explained the purpose of the code really well

siangernlow replied:
Thanks for your time and feedback! :)

underthehai commented on src/main/java/taskbot/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,46 @@
+    private String task;
+
+    /**
+     * Creates a command to make a deadline task.
```
I think there should be a line of space between the comment and the @param

underthehai commented on src/main/java/taskbot/command/DeleteCommand.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    /**
+     * @return The index of the list where the task is.
```
I think the javadoc is missing a description

underthehai reviewed (commented):
I find the documentations and naming are very clear and easy to understand.

IronBiscuit commented on src/main/java/taskbot/task/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+     *
+     * @return The time of the deadline.
+     */
+    public LocalDateTime getBy() {
```
Small nitpick, but I feel that getBy() is a rather confusing method name. Perhaps you can consider dueBy() or getDueBy()?

siangernlow replied:
Thanks for your time and feedback! Your feedback has helped me improve the code quality of my iP :)

IronBiscuit commented on src/main/java/taskbot/task/TaskList.java:
```diff
@@ -0,0 +1,311 @@
+            return "You currently have no tasks pending.";
+        }
+
+        StringBuilder sb = new StringBuilder("These are the following task(s) to complete:\n");
```
Another nitpick, perhaps a clearer name could be given to the StringBuilder object, such as "stringBuilder".

IronBiscuit commented on src/main/java/taskbot/task/TaskList.java:
```diff
@@ -0,0 +1,311 @@
+     *
+     * @return A string describing how many tasks there are in the list.
+     */
+    public String listTaskSize() {
```
Another nitpick, while listTaskSize isn't a bad method name given its function, perhaps a more appropriate name could be numberOfTasks() or getNumberOfTasks()?

IronBiscuit commented on src/main/java/taskbot/task/TaskList.java:
```diff
@@ -0,0 +1,311 @@
+        }
+        LocalDateTime dateTime = parseDateTime(parsedString[1].stripLeading());
+        // Makes a new Event task
+        Event newTask = new Event(parsedString[0], dateTime);
```
Nitpicking again! "newTask" could be a little confusing considering we have 3 unique types of tasks. Perhaps consider "event" instead?

IronBiscuit commented on src/main/java/taskbot/task/TaskList.java:
```diff
@@ -0,0 +1,311 @@
+
+        LocalDateTime dateTime = parseDateTime(parsedString[1].stripLeading());
+        // Makes a new Deadline task
+        Deadline newTask = new Deadline(parsedString[0], dateTime);
```
Same nitpick as before. Perhaps consider "deadline" or "newDeadline" instead of newTask!

IronBiscuit commented on src/test/java/taskbot/task/TaskListTest.java:
```diff
@@ -0,0 +1,217 @@
+    @Test
+    public void testListTasks() {
+        try {
+            StringBuilder expected = new StringBuilder("These are the following task(s) to complete:\n");
```
Perhaps you could rename 'expected' to 'expectedOutput'?

IronBiscuit reviewed (commented):
Great effort! 

There weren't any major naming issues, only nitpicks here and there.

siangernlow committed:
Add assertions

There may be empty strings being passed around in code, leading to
errors.

Assertions will help check for these errors.

siangernlow committed:
Improve code quality

siangernlow committed:
Merge pull request #1 from siangernlow/branch-A-Assertions

Add assertions

siangernlow committed:
Merge pull request #2 from siangernlow/branch-A-CodeQuality

Improve code quality

siangernlow committed:
Implement continuous integration

siangernlow committed:
Implement fix for Continuous Integration

siangernlow committed:
Remove unused workflows file

siangernlow committed:
Merge branch 'master' of https://github.com/siangernlow/ip

siangernlow committed:
Implement help functionality to TaskBot

siangernlow committed:
Update JavaFx text box display

Text longer than a certain length was getting clipped as the Label
class of JavaFx only supports text of limited length.

We should change the output class to the Text class, which supports
variable length text.

siangernlow committed:
Update equality check

siangernlow committed:
Implement test cases for help command

siangernlow committed:
Fix JavaDoc comments

siangernlow committed:
Fix Checkstyle issues

siangernlow committed:
Merge branch 'branch-C-Help'

siangernlow committed:
Improve code quality

siangernlow committed:
Make conversation bubbles assymmetric

Both the user and TaskBot had the same color theme for their
profile pictures.

We should display them in a different format to simulate different
parties interacting.

siangernlow committed:
Fix symbols for task completeness

Ticks and crosses indicating task completeness does not show up
properly in the application.

We should change it back to using the ascii code instead of
directly coding the symbol itself.

siangernlow committed:
Merge branch 'branch-A-BetterGui'

siangernlow committed:
Include representative screenshot of application

siangernlow committed:
Update Readme

siangernlow committed:
Set theme jekyll-theme-architect

siangernlow committed:
Update Readme

siangernlow committed:
Merge branch 'master' of https://github.com/siangernlow/ip

siangernlow committed:
Fix Checkstyle

siangernlow committed:
Make find function list tasks in order of insertion

siangernlow committed:
Fix tests
